Content edits for 22nd March 2024 workshop

alishamohanty opened the pull request:
Content edits for 22nd March 2024 workshop.
Suggested changes are elaborated in this slack [message](https://inclojureorg.slack.com/archives/CCDQALNDB/p1706184482937019).

alishamohanty committed:
Update the content file

Add
- speaker details
- structural editing
- fn example with let block
- resources link to learn clojure

alishamohanty committed:
Remove Mars Rover's quill project

PriyangaPKini commented on content.org:
```diff
@@ -31,8 +24,21 @@
+
+*** Sezal Jain
+
+# TODO 
```
Intro for Sezal is pending

alishamohanty replied:
Yeah, I haven't received a bio from her yet.

alishamohanty replied:
Have added a bio.

PriyangaPKini commented:
hey @alishamohanty, Please provide a description for this PR.

PriyangaPKini commented:
Also, are we going to share this repo with the workshop attendees? Or is this an internal reference of what we will be doing in the workshop?

alishamohanty commented:
> Also, are we going to share this repo with the workshop attendees? Or is this an internal reference of what we will be doing in the workshop?

 Yes, we will share this repo with the attendees.

qptr commented on content.org:
```diff
@@ -1,24 +1,17 @@
-  more females to learn programming, learn Clojure and do awesome
-  things that come with ability to command a computer.
+- This workshop is aimed specifically at women and non-binary people. 
+  This is a small attempt towards that. Getting more females to learn 
```
The first two sentences are disjointed.

qptr replied:
Can we reword this paragraph? I don't follow what it is trying to say

qptr commented on content.org:
```diff
@@ -31,8 +24,21 @@
 
-- https://tinyurl.com/clj-bridge-blr
--> outline.org
+- https://nilenso.com/clojurebridge
```
This link is dead. What is this supposed to point to?

alishamohanty replied:
I will redirect to the github repo instead.

qptr commented on content.org:
```diff
@@ -110,6 +116,11 @@
   learning them.
 
+- How to open a repl in VSCode 
+  - Open command using: `ctrl+shift+P` or `cmd+shift+P`
```
```suggestion
  - Open command palette using: `ctrl+shift+P` or `cmd+shift+P`
```

qptr commented on content.org:
```diff
@@ -110,6 +116,11 @@
 - Let's start a REPL. and we can type and evaluate things as we go on
   learning them.
 
+- How to open a repl in VSCode 
```
```suggestion
- How to open a REPL in VSCode
```

qptr commented on content.org:
```diff
@@ -239,6 +250,25 @@ Evaluate some normal functions in your REPL
 (,+ 1 2)
 #+END_SRC
 
+**** Structural Editing
```
Solid section

qptr replied:
Can we move this part to below the whole Expressions section? It breaks it up while reading so the notation part of Expressions is below Strutural Editing. I know this is where structural editing becomes relavant. Maybe we can add a note saying `See below: Structural Editing` so it reads smoother.

Right now, it looks like Structural editing, Infix notation, Prefix notation are all subheadings under Expressions.

Some kind of formatting to set it apart form the learning material would help with this too, something like [this](https://github.blog/changelog/2023-12-14-new-markdown-extension-alerts-provide-distinctive-styling-for-significant-content/)? Leaving it upto you

alishamohanty replied:
I have moved "Structural Editing" section post "Expression" section.

qptr commented on content.org:
```diff
@@ -239,6 +250,25 @@ Evaluate some normal functions in your REPL
+rather than its text representation.
+It allows for operations like moving, wrapping or editing code blocks as discrete units.
+
+Plugins that enable structural editing:
+
+Visual studio code: Calva (https://calva.io/)
+Intellij: Cursive (https://cursive-ide.com/)
```
Bullets so its clearer to see
```suggestion
Plugins that enable structural editing:

- Visual Studio Code: Calva (https://calva.io/)
- Intellij: Cursive (https://cursive-ide.com/)
```

qptr commented on content.org:
```diff
@@ -611,7 +649,7 @@ List of basic types:
 #+BEGIN_SRC clojure
 (first [5 10 15])
-(nth [5 10 15] 0)
+;; TODO write using nth function?
```
I think we can keep this around. Seems like it's just to reinforce how nth works. Applying something we learned right away in a different context.

alishamohanty replied:
Yes, but we will ask this to the audience. Thats why I have removed the solution.

alishamohanty replied:
Yes, but we will ask this to the audience. Thats why I have removed the solution.

qptr commented on resources.org:
```diff
@@ -0,0 +1 @@
+Clojure learning material;s curated list by Srihari: https://gist.github.com/ssrihari/0bf159afb781eef7cc552a1a0b17786f
```
Stray semicolon

qptr replied:
Think this page can use formatting
```suggestion
# More Clojure learning resources

[Curated list by Srihari](https://gist.github.com/ssrihari/0bf159afb781eef7cc552a1a0b17786f)
```

qptr reviewed (changes requested):
Mostly looks solid.

Why have we removed all the exercises from this repo? What's the plan?

qptr commented on content.org:
```diff
@@ -31,8 +24,21 @@
+
+** Speakers
+
+*** Soumya Maithani
```
Typo: Somya

alishamohanty commented:
> Mostly looks solid.
> 
> Why have we removed all the exercises from this repo? What's the plan?

We haven't removed any exercises. They still exist in the exercises folder. We have only removed the mars rover problem which is not a part of the curriculum due to limited time constraint. We will only solve Land of Wizards problem during the workshop.

sezaljain commented on content.org:
```diff
@@ -1,24 +1,16 @@
-  more females to learn programming, learn Clojure and do awesome
-  things that come with ability to command a computer.
+- This workshop is specifically designed for women and non-binary individuals. 
+It's a small effort to encourage more females to learn programming, particularly
```
'more females' -> them

sezaljain replied:
acutally, I would suggest to keep the original content

sezaljain commented on content.org:
```diff
@@ -284,6 +297,26 @@ Evaluate some normal functions in your REPL
+| Slurp Forward              | ctrl+alt+right         | ctrl+alt+. | ctrl+opt+right          |
+| Barf Forward               | ctrl+alt+left          | ctrl+alt+, | ctrl+opt+left           |
+| Raise Sexp                 |       ctrl+alt+p  ctrl+alt+r        | ctrl+opt+p  ctrl+opt+r  |
+| Drag sexp forward/backward |           alt+up/down arrows        | opt+up/down arrows      |
```
try fixing the table here by writing something like
| Raise Sexp                 |       ctrl+alt+p  ctrl+alt+r       || ctrl+opt+p  ctrl+opt+r  |

alishamohanty committed:
Add paredit link in structural editing

- Add compand to evaluate a sexp in REPL section
- fix the view of paredit table
- undo the removal of "why? clojurebridge" in the intro

alishamohanty committed:
Fix formatting in resources file

Update the type of file from org to md
